Apps/GatewayApi/src/Services/WebAlertService.cs

```diff
@@ -0,0 +1,125 @@
+    using System.Threading.Tasks;
+    using AutoMapper;
+    using HealthGateway.Common.Constants;
+    using HealthGateway.Common.Models.PHSA;
+    using HealthGateway.Common.Services;
+    using HealthGateway.GatewayApi.Api;
+    using HealthGateway.GatewayApi.Models;
+    using HealthGateway.GatewayApi.Models.Phsa;
+    using Microsoft.Extensions.Logging;
+    using Refit;
+
+    /// <summary>
+    /// The Patient data service.
+    /// </summary>
+    public class WebAlertService : IWebAlertService
+    {
+        private readonly ILogger<WebAlertService> logger;
+        private readonly IPersonalAccountsService personalAccountsService;
+        private readonly IWebAlertApi webAlertApi;
+        private readonly IMapper autoMapper;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebAlertService"/> class.
+        /// </summary>
+        /// <param name="logger">The injected logger.</param>
+        /// <param name="personalAccountsService">The injected personal accounts service.</param>
+        /// <param name="webAlertApi">The injected Refit API.</param>
+        /// <param name="autoMapper">The injected automapper.</param>
+        public WebAlertService(ILogger<WebAlertService> logger, IPersonalAccountsService personalAccountsService, IWebAlertApi webAlertApi, IMapper autoMapper)
+        {
+            this.logger = logger;
+            this.personalAccountsService = personalAccountsService;
+            this.webAlertApi = webAlertApi;
+            this.autoMapper = autoMapper;
+        }
+
+        private static ActivitySource Source { get; } = new(nameof(WebAlertService));
+
+        /// <inheritdoc/>
+        public async Task<IList<WebAlert>> GetWebAlertsAsync(string hdid)
+        {
+            using Activity? activity = Source.StartActivity();
+            this.logger.LogDebug("Retrieving web alerts from PHSA.");
+            try
+            {
+                PersonalAccount personalAccount = await this.personalAccountsService.GetPatientAccountAsync(hdid).ConfigureAwait(true);
+                string pid = personalAccount.PatientIdentity.Pid.ToString();
+
+                IList<PhsaWebAlert> phsaWebAlerts = await this.webAlertApi.GetWebAlertsAsync(pid).ConfigureAwait(true);
+                IList<WebAlert> webAlerts = this.autoMapper.Map<IEnumerable<PhsaWebAlert>, IList<WebAlert>>(
+                    phsaWebAlerts.Where(a => a.ExpirationDateTimeUtc > DateTime.UtcNow && a.ScheduledDateTimeUtc < DateTime.UtcNow));
+                this.logger.LogDebug("Finished retrieving web alerts from PHSA.");
+                return webAlerts;
+            }
+            catch (Exception e) when (e is ApiException or HttpRequestException)
+            {
+                this.logger.LogError(e, "Error retrieving web alerts from PHSA.");
+                throw new Common.Data.ErrorHandling.ApiException(ErrorMessages.RecordsNotAvailable, HttpStatusCode.BadGateway, nameof(WebAlertService));
```

Comment:
Using BadGateway, has this been discussed and agreed upon by the team?

Reply:
We shouldn't have this catch at all and instead let the problem details middleware wrap the true exception instead of masked one.

Reply:
Removed try/catch block.